nilearn/image/tests/test_resampling.py

```diff
@@ -145,6 +145,36 @@ def test_downsample():
                                        rot_img.get_data()[:x, :y, :z, ...])
 
 
+def test_resampling_fill_value():
+    """ Test resampling with a non-zero fill value
+    """
+    prng = np.random.RandomState(10)
+
+    data_3d = prng.rand(1, 4, 4)
+    data_4d = prng.rand(1, 4, 4, 3)
+
+    angle = np.pi/4
+    rot = rotation(0, angle)
+
+    # Try a few different fill values
+    for data in [data_3d, data_4d]:
+        for val in (-3, -1.7537, 0, 0.666, 11):
```

Comment:
In the interest of test running time, could we make this list smaller, for instance keeping (-3, 0.666, 11).

Reply:
Actually, I am not sure that I see the point of testing with more than 2 values.

Reply:
You could do two checks, one explicitly specifying a non-zero value, and one leaving the default and checking you get 0

Reply:
ok, so, I went with large negative (<-1) and large positive (>1)  values with decimals, which test that clipping is indeed not performed on the filling value and possibly rounding / type errors. I also added a test of the default value (0) as suggested.